packages/engine/Source/Core/addAll.js

```diff
@@ -0,0 +1,41 @@
+import defined from "./defined.js";
+
+/**
+ * Adds all elements from the given source array to the given target array.
+ *
+ * If the `source` is `undefined`, then nothing will be done. Otherwise,
+ * this has the same semantics as
+ * ```
+ * for (const s of source) target.push(s);
+ * ```
+ * but is usually more efficient than a `for`-loop, and does not put the
+ * elements of the source on the stack, as it would be done with the
+ * spread operator or when using `target.push.apply(source)`.
+ *
+ * @function
+ * @private
+ *
+ * @param {Array|undefined} source The source array
+ * @param {Array} target The target array
+ *
+ * @example
+ * const target = [ 0, 1, 2 ];
+ * const source = [ 3, 4, 5 ];
+ * Cesium.addAll(source, target);
+ * // The target is now [ 0, 1, 2, 3, 4, 5 ]
```

Comment:
Nitpick on semantics, but this time I do think it actually matters. When I first glanced at this code I expected the order of the arguments to be reversed: `addAll(target, source)`.
This is largely because I am comparing this to similar functions of `target.push(...source)` or `target.concat(source)` etc. In all the builtin array functions the "result" is on the left and the "source" is on the right. 
Another way I think about it is that this is replacing the syntax `[...target, ...source]` to put all the target elements first and the source elements second. So the target variable should be the first argument. Even in this example you're doing `addAll([ 3, 4, 5 ], [ 0, 1, 2 ]) = [ 0, 1, 2, 3, 4, 5 ]`
(Maybe "source" and "target" are the wrong names in the first place)

I think swapping them also aligns better with the idea that the "source" could be `null`, `undefined` or `[]`. Then the first argument is always defined even if the second only sometimes is.

Reply:
I also hesitated here. Leaving the source as the last parameter would allow omitting it, as in
`addAll(target);`
which is probably not ideal. 

But in general, I'm open to changing it, and embarassingly have to point out that commit https://github.com/CesiumGS/cesium/pull/12361/commits/c00243a948e5a8dd99a0fe6bdd5d91d5d60bbbb2 ...

Reply:
From my all-time favorite [presentation on API design](https://www.youtube.com/watch?v=aAb7hSCtvGw):

![Cesium Parameter order](https://github.com/user-attachments/assets/3ecbc2ee-5501-4d4b-bac9-b3226d09168c)

Reply:
Swapped the parameters. Fortunately, someone [posted the Regex for that](https://stackoverflow.com/a/71259650), which is to ...
replace `(addAllToArray)\(([^,]+), *([^,]+)\)`
with `$1($3, $2)`